diff --git a/legion/engine/core/compute/program.cpp b/legion/engine/core/compute/program.cpp
index 2abd681a..1786c095 100644
--- a/legion/engine/core/compute/program.cpp
+++ b/legion/engine/core/compute/program.cpp
@@ -73,6 +73,13 @@ namespace legion::core::compute {
         if (ret != CL_SUCCESS)
         {
             log::error("clBuildProgram failed");
+
+            size_t length;
+            char buffer[8192];
+
+            clGetProgramBuildInfo(m_program,device,CL_PROGRAM_BUILD_LOG,sizeof(buffer),buffer,&length);
+            buffer[length] = NULL;
+            log::warn("BUILD LOG:\n{}",buffer);
         }
 
         /*}*/
